GitCommands/Git/GitModule.cs

```diff
@@ -2313,8 +2313,14 @@ public string GetSetting(string setting)
             return LocalConfigFile.GetValue(setting);
         }
 
-        public string GetEffectiveSetting(string setting)
+        public string GetEffectiveSetting(string setting, bool useGit = false, bool useCache = false)
```

Comment:
Use a separate function instead?

Reply:
Yes please. Let's call it `GetEffectiveGitSetting()`. Cache by default until we have a real use case that requires no caching.